src/Symfony/Component/Form/Tests/Extension/Validator/Type/FormTypeValidatorExtensionTest.php

```diff
@@ -158,4 +161,136 @@ protected function createForm(array $options = [])
+                    return $organizationMetadata;
+                }
+
+                if (Form::class === $classOrObject || $classOrObject instanceof Form) {
+                    return $formMetadata;
+                }
+
+                return new ClassMetadata(\is_string($classOrObject) ? $classOrObject : \get_class($classOrObject));
+            });
+
+        $validator = Validation::createValidatorBuilder()
+            ->setMetadataFactory($metadataFactory)
+            ->getValidator();
+
+        $submitData = [
+            'authors' => [
+                0 => [
+                    'firstName' => '', // Fires a Not Blank Error
+                    'lastName' => 'lastName1',
+                ],
+                // key "1" could be missing if we add 4 blank form entries and then remove it.
+                2 => [
+                    'firstName' => '', // Fires a Not Blank Error
+                    'lastName' => 'lastName3',
+                ],
+                3 => [
+                    'firstName' => '', // Fires a Not Blank Error
+                    'lastName' => 'lastName3',
+                ],
+            ],
+        ];
+
+        $form = Forms::createFormFactoryBuilder()
+            ->addExtension(new ValidatorExtension($validator))
+            ->getFormFactory()
+            ->create(FormTypeTest::TESTED_TYPE, new Organization([]), [
+                'data_class' => Organization::class,
+                'by_reference' => false,
+            ])
+            ->add('authors', CollectionTypeTest::TESTED_TYPE, [
+                'entry_type' => AuthorType::class,
+                'allow_add' => true,
+                'allow_delete' => true,
+            ])
+        ;
+
+        $form->submit($submitData);
+
+        //Form behaves right (...?). It has index 0, 2 and 3 (1 has been removed)
+        $this->assertTrue($form->get('authors')->has('0'));
+        $this->assertFalse($form->get('authors')->has('1'));
+        $this->assertTrue($form->get('authors')->has('2'));
+        $this->assertTrue($form->get('authors')->has('3'));
+
+        //Form does have 3 not blank errors
+        $errors = $form->getErrors(true);
+        $this->assertCount(3, $errors);
+
+        //But errors property paths are messing up
+        $errorPaths = [
+            $errors[0]->getCause()->getPropertyPath(),
+            $errors[1]->getCause()->getPropertyPath(),
+            $errors[2]->getCause()->getPropertyPath(),
+        ];
+
+        $this->assertContains('data.authors[0].firstName', $errorPaths);
+        $this->assertContains('data.authors[1].firstName', $errorPaths);
+        $this->assertContains('data.authors[2].firstName', $errorPaths);
+        $this->assertNotContains('data.authors[3].firstName', $errorPaths);
+
+        //In fact, root form should NOT contain errors but it does
+        $this->assertCount(1, $form->getErrors(false));
+
+        //Let's do this again, but with "keep_as_list" option set to true
```

Comment:
Can you create another test method instead of grouping two tests in the same one please?

Reply:
We also need a rebase and to update the changelog.
Please tell us if you want to do it or if we should take over, thank you.